website/docs/r/privatelink_endpoint_service_data_federation_online_archive.markdown

```diff
@@ -0,0 +1,53 @@
+---
+layout: "mongodbatlas"
+page_title: "MongoDB Atlas: mongodbatlas_privatelink_endpoint_service_data_federation_online_archive"
+sidebar_current: "docs-mongodbatlas-resource-privatelink-endpoint-service-data-federation-online-archive"
+description: |-
+    Provides a Privatelink Endpoint Service Data Federation Online Archive resource.
+---
+
+# Resource: mongodbatlas_privatelink_endpoint_service_data_federation_online_archive
+
+`mongodbatlas_privatelink_endpoint_service_data_federation_online_archive` provides a Private Endpoint Service resource for Data Federation and Online Archive. The resource allows you to create and manage a private endpoint for Federated Database Instances and Online Archives to the specified project.
+
+-> **NOTE:** Groups and projects are synonymous terms. You may find `groupId` in the official documentation.
+
+## Example Usage
+
+```terraform
+resource "mongodbatlas_project" "atlas-project" {
+  org_id = var.atlas_org_id
+  name   = var.atlas_project_name
+}
+
+resource "mongodbatlas_privatelink_endpoint_service_data_federation_online_archive" "test" {
+  project_id = mongodbatlas_project.atlas-project.id
+  endpoint_id = "<PRIVATE-ENDPOINT-SERVICE-ID>"
+  provider_name = "AWS"
+  type = "DATA_LAKE"
+  comment = "Test"
+}
+
+## Argument Reference
+
+* `project_id` (Required) - Unique 24-hexadecimal digit string that identifies your project. 
+* `endpoint_id` (Required) - Unique 22-character alphanumeric string that identifies the private endpoint. See [Atlas Data Lake supports Amazon Web Services private endpoints using the AWS PrivateLink feature](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Data-Federation/operation/createDataFederationPrivateEndpoint:~:text=Atlas%20Data%20Lake%20supports%20Amazon%20Web%20Services%20private%20endpoints%20using%20the%20AWS%20PrivateLink%20feature.).
+* `provider_name` (Required) - Human-readable label that identifies the cloud service provider. 
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `type` - Human-readable label that identifies the resource type associated with this private endpoint.
```

Comment:
Since this only has a type of DATA_LAKE and since the name doesn't even match the product name anymore can we consider setting this for the user, i.e. it's not possible to be another value so why have the user set it?  I think this, hopefully, simple change will make this easier across the board.

Reply:
sure, it makes sense

Reply:
Updated. I set the `type` field to `Computed` also for the resource and updated the logic to set its value during the CREATE operation. Thanks

Reply:
Thank you @andreaangiolillo!  Will resolve this but I left a few comments where we'll need to remove it from the docs examples since now it's not user set-able.